src/app/fdctl/run/topos/fd_frankendancer.c

```diff
@@ -81,15 +80,26 @@ fd_topo_frankendancer( config_t * config ) {
 
   ushort parsed_tile_to_cpu[ FD_TILE_MAX ];
   for( ulong i=0UL; i<FD_TILE_MAX; i++ ) parsed_tile_to_cpu[ i ] = USHORT_MAX; /* Unassigned tiles will be floating. */
-  ulong affinity_tile_cnt = fd_tile_private_cpus_parse( config->layout.affinity, parsed_tile_to_cpu );
+
+  int is_auto_affinity = !strcmp( config->layout.affinity, "auto" );
+  int is_agave_auto_affinity = !strcmp( config->layout.agave_affinity, "auto" );
+  int is_bench_auto_affinity = !strcmp( config->development.bench.affinity, "auto" );
+
+  if( FD_UNLIKELY( is_auto_affinity != is_agave_auto_affinity ||
+                   is_auto_affinity != is_bench_auto_affinity ) ) {
+    FD_LOG_ERR(( "The CPU affinity string in the configuration file under [layout.affinity], [layout.agave_affinity], and [development.bench.affinity] must all be set to 'auto' or all be set to a specific CPU affinity string." ));
+  }
+
+  ulong affinity_tile_cnt = 0UL;
+  if( FD_LIKELY( !is_auto_affinity ) ) affinity_tile_cnt = fd_tile_private_cpus_parse( config->layout.affinity, parsed_tile_to_cpu );
 
   ulong tile_to_cpu[ FD_TILE_MAX ];
```

Comment:
In the auto case, this should be zero-ed or set to USHORT_MAX so that when we read it below, it doesn't get garbage. I know the garbage will be overwritten, but that still seems safer.